File: tutorials/testing/c++/tut_graph_p2.cpp
#include <string>
#include "Bridges.h"
#include "GraphAdjList.h"

using namespace std;
using namespace bridges;

// Part 2 of this tutorial will demonstrate how to style the edges and links of 
// an adjacency list based graph using BRIDGES and display it.
// Relevant Classes: GraphAdjList class, ElementVisualizer, 
// LinkVisualizer, Color. We will use the same graph as in  Part 1

int main(int argc, char **argv) {
	// create Bridges object, set credentials
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");

	// set a title 
	bridges.setTitle("A Simple Adjacency list based Graph Example.");

	// set  description
	bridges.setDescription("Demonstrate styling graph nodes and links with visual attributes");

	// create a graph with key values of type string. See the
	// the documentation for GraphAdjList for its full capabilities
	GraphAdjList<string> graph;

	// first create a few vertices 

	// create some actor names to be added to the graph
	string 	kevin_bacon = "Kevin Bacon", 
			denzel_washington = "Denzel Washington",
			morgan_freeman = "Morgan Freeman", 
			tom_cruise = "Tom Cruise", 
			angelina_jolie = "Angelina Jolie", 
			amy_adams = "Amy Adams", 
			brad_pitt = "Brad Pitt"; 
	// add them to the graph
	graph.addVertex(kevin_bacon);
	graph.addVertex(denzel_washington);
	graph.addVertex(morgan_freeman);
	graph.addVertex(tom_cruise);
	graph.addVertex(angelina_jolie);
	graph.addVertex(amy_adams);
	graph.addVertex(brad_pitt);

	// add edges
	graph.addEdge(kevin_bacon, denzel_washington);
	graph.addEdge(kevin_bacon, morgan_freeman);
	graph.addEdge(kevin_bacon, angelina_jolie);
	graph.addEdge(amy_adams, angelina_jolie);
	graph.addEdge(tom_cruise, amy_adams);
	graph.addEdge(angelina_jolie, brad_pitt);
	graph.addEdge(brad_pitt, denzel_washington);

	// style  the nodes
	
	// distinguish the male and female actors by color
    graph.getVertex(kevin_bacon)->setColor("limegreen");
    graph.getVertex(brad_pitt)->setColor("limegreen");
    graph.getVertex(morgan_freeman)->setColor("limegreen");
    graph.getVertex(denzel_washington)->setColor("limegreen");
    graph.getVertex(tom_cruise)->setColor("limegreen");
    graph.getVertex(angelina_jolie)->setColor("crimson");
    graph.getVertex(amy_adams)->setColor("crimson");

	// shape
    graph.getVertex(brad_pitt)->setShape(SQUARE);
    graph.getVertex(angelina_jolie)->setShape(SQUARE);

	// opacity
    graph.getVertex(tom_cruise)->setOpacity(0.3f);

	// style the links
	graph.getLinkVisualizer(angelina_jolie, brad_pitt)->setColor("orange");
	graph.getLinkVisualizer(angelina_jolie, brad_pitt)->setLabel("Dated/Married");
	graph.getLinkVisualizer(angelina_jolie, brad_pitt)->setThickness(5.0f);

	// provide BRIDGES the  handle to the tree structure
	bridges.setDataStructure(&graph);

	// Visualize the graph
	bridges.visualize();

	return 0;
}

File: tutorials/testing/python/tut_graph_p2.py
from bridges.data_src_dependent import *
from bridges.graph_adj_list import *
from bridges.bridges import *
import sys


def main():
    # This tutorial will demonstrate how to build an adjacency list based graph
    # using BRIDGES and display it.
    # References: GraphAdjList class

    # create the Bridges object, set credentials

    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")

    # set a title 
    bridges.set_title("A Simple Adjacency list based Graph Example.")

    # set  description
    bridges.set_description("Demonstrate styling graph nodes and links with visual attributes")

    # create an adjacency list based graph
    graph = GraphAdjList()

    # create some actor names to be added to the graph
    kevin_bacon = "Kevin Bacon" 
    denzel_washington = "Denzel Washington"
    morgan_freeman = "Morgan Freeman"
    tom_cruise = "Tom Cruise"
    angelina_jolie = "Angelina Jolie"
    amy_adams = "Amy Adams" 
    brad_pitt = "Brad Pitt"

    # add them to the graph
    graph.add_vertex(kevin_bacon)
    graph.add_vertex(denzel_washington)
    graph.add_vertex(morgan_freeman)
    graph.add_vertex(tom_cruise)
    graph.add_vertex(angelina_jolie)
    graph.add_vertex(amy_adams)
    graph.add_vertex(brad_pitt)

    # add edges
    graph.add_edge(kevin_bacon, denzel_washington)
    graph.add_edge(kevin_bacon, morgan_freeman)
    graph.add_edge(kevin_bacon, angelina_jolie)
    graph.add_edge(amy_adams, angelina_jolie)
    graph.add_edge(tom_cruise, amy_adams)
    graph.add_edge(angelina_jolie, brad_pitt)
    graph.add_edge(brad_pitt, denzel_washington)

    # style  the nodes
    
    # distinguish the male and female actors by color
    graph.get_vertex(kevin_bacon).color = "limegreen"
    graph.get_vertex(brad_pitt).color = "limegreen"
    graph.get_vertex(morgan_freeman).color = "limegreen"
    graph.get_vertex(denzel_washington).color = "limegreen"
    graph.get_vertex(tom_cruise).color = "limegreen"
    graph.get_vertex(angelina_jolie).color = "crimson"
    graph.get_vertex(amy_adams).color = "crimson"

    # shape
    graph.get_vertex(brad_pitt).shape = "square"
    graph.get_vertex(angelina_jolie).shape = "square"

    # opacity
    graph.get_vertex(tom_cruise).opacity = 0.3

    # style the links
    graph.get_link_visualizer(angelina_jolie, brad_pitt).color = "orange"
    graph.get_link_visualizer(angelina_jolie, brad_pitt).label = "Dated/Married"
    graph.get_link_visualizer(angelina_jolie, brad_pitt).thickness = 5.0

    # Pass the graph object to BRIDGES
    bridges.set_data_structure(graph)

    # visualize the graph
    bridges.visualize()


if __name__ == "__main__":
    main()

